.github/workflows/build.yml

```diff
@@ -104,29 +48,359 @@ jobs:
         run: west update
       - name: Export Zephyr CMake package (west zephyr-export)
         run: west zephyr-export
-      - name: Prepare variables
-        id: variables
-        run: |
-          SHIELD_ARG=
-          ARTIFACT_NAME="${{ matrix.board }}"
-
-          if [ -n "${{ matrix.shield }}" ]; then
-            SHIELD_ARG="-DSHIELD=${{ matrix.shield }}"
-            ARTIFACT_NAME="${ARTIFACT_NAME}-${{ matrix.shield }}"
-          fi
-
-          ARTIFACT_NAME="${ARTIFACT_NAME}-zmk"
-
-          echo ::set-output name=shield-arg::${SHIELD_ARG}
-          echo ::set-output name=artifact-name::${ARTIFACT_NAME}
-      - name: Build (west build)
-        run: west build -s app -b ${{ matrix.board }} -- ${{ steps.variables.outputs.shield-arg }} ${{ matrix.cmake-args }}
-      - name: Archive artifacts
-        if: ${{ !matrix.skip-archive }}
-        uses: actions/upload-artifact@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v2
         with:
-          name: "${{ steps.variables.outputs.artifact-name }}"
-          path: |
-            build/zephyr/zmk.hex
-            build/zephyr/zmk.uf2
-        continue-on-error: true
+          node-version: '14.x'
+      - name: Install @actions/artifact
+        run: npm install @actions/artifact
+      - name: Build and upload artifacts
+        uses: actions/github-script@v4
+        id: boards-list
+        with:
+          script: |
+            const fs = require('fs');
+            const artifact = require('@actions/artifact');
+            const artifactClient = artifact.create();
+
+            const execSync = require('child_process').execSync;
+            
+            const buildShieldArgs = JSON.parse(`${{ matrix.shieldArgs }}`);
+
+            let error = false;
+
+            for (const shieldArgs of buildShieldArgs) {
+              try {
+                const output = execSync(`west build -s app -p -b ${{ matrix.board }} -- ${shieldArgs.shield ? '-DSHIELD=' + shieldArgs.shield : ''} ${shieldArgs['cmake-args'] || ''}`);
+
+                console.log(`::group::${{ matrix.board}} ${shieldArgs.shield} Build`)
+                console.log(output.toString());
+                console.log('::endgroup::');
```

Comment:
We should put this endgroup into a `finally` block so it there are any exceptions logged in the `catch` it's part of the group.